homeassistant/components/squeezebox/media_player.py

```diff
@@ -587,4 +594,37 @@ async def async_browse_media(self, media_content_type=None, media_content_id=Non
             "search_id": media_content_id,
         }
 
-        return await build_item_response(self._player, payload)
+        return await build_item_response(self, payload)
+
+    async def async_get_browse_image(self, browse_image):
+        """Get album art from Squeezebox server."""
+        if browse_image.startswith(self._internal_artwork_url):
+            try:
+                session = async_get_clientsession(self.hass)
+                with async_timeout.timeout(5):
+                    response = await session.get(browse_image)
```

Comment:
i wonder if theres a way to provide a helper to get image type and content so we dont have multiple variabta of this code. then the implementors only need to pass a url through helper or if the integration gives blobs it can directly pass such through

Reply:
This is a good idea. There is a helper in `media_player._async_fetch_image`. I was reluctant to use it here because it caches the images in the server's memory, which might not be a good idea, because we will exceed the cache size if we more than a few responsive items, and it would displace the currently playing album art which makes more sense for the cache (since it might be retrieved repeatedly). 

I think the solution is to add an optional `cache` parameter to `_async_fetch_image` that allows disabling the cache for a particular query. That way we don't duplicate the code that does the actual retrieval.

Reply:
sounds reasonable. thanks for the follow up